relay-config/src/config.rs

```diff
@@ -2003,6 +2004,75 @@ impl Config {
         self.values.processing.max_rate_limit.map(u32::into)
     }
 
+    /// Returns the most permissible [`AggregatorConfig`] based on the values of all the configured aggregators.
+    pub fn permissive_aggregator_config(&self) -> AggregatorConfig {
+        let aggregators: Vec<AggregatorConfig> = self
+            .values
+            .secondary_aggregators
+            .iter()
+            .map(|agg| &agg.config.aggregator)
+            .chain(std::iter::once(&self.values.aggregator.aggregator))
+            .cloned()
+            .collect();
+
+        // The unwraps are safe, as it's only 'None' if the vector is empty.
+        let bucket_interval = aggregators
+            .iter()
+            .min_by_key(|agg| agg.bucket_interval)
+            .unwrap()
+            .bucket_interval;
+
+        for agg in &aggregators {
+            if agg.bucket_interval % bucket_interval != 0 {
+                relay_log::error!("buckets don't align");
+            }
+        }
+
+        let max_secs_in_past = aggregators
+            .iter()
+            .max_by_key(|agg| agg.max_secs_in_past)
+            .unwrap()
+            .bucket_interval;
+
+        let max_secs_in_future = aggregators
+            .iter()
+            .max_by_key(|agg| agg.max_secs_in_future)
+            .unwrap()
+            .bucket_interval;
+
+        let max_name_length = aggregators
+            .iter()
+            .max_by_key(|agg| agg.max_name_length)
+            .unwrap()
+            .bucket_interval as usize;
+
+        let max_tag_key_length = aggregators
+            .iter()
+            .max_by_key(|agg| agg.max_tag_key_length)
+            .unwrap()
+            .bucket_interval as usize;
+
+        let max_tag_value_length = aggregators
+            .iter()
+            .max_by_key(|agg| agg.max_tag_value_length)
+            .unwrap()
+            .bucket_interval as usize;
+
+        // None means 'no limit'.
+        let max_project_key_bucket_bytes = None;
```

Comment:
I think there's a simpler way to code this:

1. Make a private `aggregators()` helper method. Can return an `impl Iterator`, does not have to be a vector.
2. Write a for loop that collects the min/max values
3. Write a second for loop that checks that all bucket intervals align with the min bucket interval

Reply:
is the concern performance or readability ? I think your suggestion would be faster but im not sure if the code would be any simpler, or maybe im not visualizing it correctly